library/core/src/main/java/com/google/android/exoplayer2/text/ttml/TtmlNode.java

```diff
@@ -409,17 +409,29 @@ private void applyStyleToOutput(
     if (resolvedStyle != null) {
       TtmlRenderUtil.applyStylesToSpan(
           text, start, end, resolvedStyle, parent, globalStyles, verticalType);
-      if (resolvedStyle.getShearPercentage() != TtmlStyle.UNSPECIFIED_SHEAR && TAG_P.equals(tag)) {
-        // Shear style should only be applied to P nodes
-        // https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-shear
-        // The spec doesn't specify the coordinate system to use for block shear
-        // however the spec shows examples of how different values are expected to be rendered.
-        // See: https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-shear
-        // https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-fontShear
-        // This maps the shear percentage to shear angle in graphics coordinates
-        regionOutput.setShearDegrees((resolvedStyle.getShearPercentage() * -90) / 100);
+      if (TAG_P.equals(tag)) {
+        if (resolvedStyle.getShearPercentage() != TtmlStyle.UNSPECIFIED_SHEAR) {
+          // Shear style should only be applied to P nodes
+          // https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-shear
+          // The spec doesn't specify the coordinate system to use for block shear
+          // however the spec shows examples of how different values are expected to be rendered.
+          // See: https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-shear
+          // https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-fontShear
+          // This maps the shear percentage to shear angle in graphics coordinates
+          regionOutput.setShearDegrees((resolvedStyle.getShearPercentage() * -90) / 100);
+        }
+        if (resolvedStyle.getMultiRowAlign() != null) {
+          // Multi-row align should only be applied to P nodes
+          // https://www.w3.org/TR/ttml-imsc1.1/#text-multiRowAlign
+          regionOutput.setMultiRowAlignment(resolvedStyle.getMultiRowAlign());
+        }
+      }
+
+      if (resolvedStyle.getTextAlign() != null) {
+        // Only set text alignment if defined in current node, otherwise could overwrite
+        // inherited text alignment
+        regionOutput.setTextAlignment(resolvedStyle.getTextAlign());
```

Comment:
This looks strange to me - could you explain a bit more why it's needed? (and why it's only needed for text alignment, and none of the other properties)

Reply:
(by 'this' I meant the `if`, even though I commented on the `set` line...)

I tried removing the `if` (i.e. reverting to the previous code that just always calls `setTextAlignment()`) and ran the demo file you provided and it seemed to work fine for the I.3 case you screenshotted but not the I.14 case - is that what this enables to work?

Reply:
Ah is this an existing bug in the TTML implementation?

I can reproduce with this test data:
```
<div>
   <p begin="70s" end="78s" tts:textAlign="start"><span tts:fontSize="50%">Start</span> alignment</p>
</div>
```

And this expectation fails because `textAlignment` is actually null:
```
assertThat(seventhCue.textAlignment).isEqualTo(Layout.Alignment.ALIGN_NORMAL);
```

I'm still a bit suspicious that only `textAlignment` needs special treatment like this - if you can help me understand that I'd appreciate it, I'll keep playing around on my end too.

Reply:
Ah OK I see it's completely the opposite way round: `textAlignment` is the only property in this bit of the code that isn't already null-checked - sorry for being a bit slow here.

Shouldn't it also only be applied to `<p>` nodes (like shear above, and now multiRowAlign)?
https://www.w3.org/TR/2018/REC-ttml2-20181108/#style-attribute-textAlign

I think this probably deserves to be in a separate change as a 'bug fix' - would you like to pull it out into a separate PR or shall I just make it as a commit before merging this on top?

Reply:
Created https://github.com/google/ExoPlayer/pull/8877 for the text alignment bug